Live activities (#4908)
braze-inc/braze-docs: _docs/_api/endpoints/messaging/live_activity/update.md

---
hidden: true
nav_title: "POST: Update Live Activity"
article_title: "POST: Update Live Activity"
search_tag: Endpoint
page_order: 1

layout: api_page
page_type: reference
description: "This article outlines details about the Update Live Activity endpoint."

---
{% api %}
# Update Live Activity
{% apimethod post %}
/messages/live_activity/update
{% endapimethod %}

> Use this endpoint to update and end Live Activities displayed by your iOS app.

Reviewer: ```suggestion
Use this endpoint to update and end [Live Activities](({{site.baseurl}}/developer_guide/platform_integration_guides/swift/live_activities/)) displayed by your iOS app.
```

> Use this endpoint to handle updates for Live Activities displayed by your iOS app.

Reviewer 2: Use this endpoint to update and end Live Activities displayed by your iOS app

> Before using this endpoint, you must register an activity with the Braze Swift SDK using the [`launchActivity`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/launchactivity(pushtokentag:activity:fileid:line:)) method. Required request parameters will be defined during this step. See the [Live Activities article]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/live_activities/) for more information on registration.

Reviewer: ```suggestion
Before using this endpoint, you must register an activity with the Braze Swift SDK using the [`launchActivity`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/launchactivity(pushtokentag:activity:fileid:line:)) method. Required request parameters will be defined during this step. Refer to [Live Activities]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/live_activities/) for more information on registration.
```

## Rate limit

<!--What is the rate limit? -->

{% multi_lang_include rate_limits.md endpoint='default' category='message endpoints' %}

> 

Reviewer: ```suggestion

## Request body

```
Content-Type: application/json
Authorization: Bearer YOUR-REST-API-KEY
```

```json
{
   "app_id": (required, string) App API identifier retrieved from the Developer Console.,
   "activity_id": (required, string) When you register your Live Activity using launchActivity, you use the pushTokenTag parameter to name the Activity’s push token to a custom string. Set activity_id to this custom string to define which Live Activity you want to update.,
   "content_state": (required, object) You define the ContentState parameters when you create your Live Activity. Pass the updated values for your ContentState using this object. The format of this request must match the shape you initially defined,
   "end_activity": (optional, boolean) If true, this request ends the Live Activity,
   "dismissal_date": (optional, datetime in ISO-8601 format) the time to remove the Live Activity from the user’s UI. If this time is in the past, the Live Activity will be removed immediately,
   "stale_date": (optional, datetime in ISO-8601 format) when the Live Activity content is marked as outdated in the user’s UI,
   "notification": (optional, object ) include an apple_push object to define a push notification that will also be displayed with this update.
 }
```

```

## Request parameters

<!--Please review for accuracy-->

| Parameter | Required | Data Type | Description |
|---|---|---|---|
| `app_id` | Required | String | App [API identifier]({{site.baseurl}}/api/identifier_types/#the-app-identifier) retrieved from the **Developer Console**.  |

> | `activity_id` | Required | String | When you register your Live Activity using [`launchActivity`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class), you use the `pushTokenTag` parameter to name the Activity's push token to a custom string. Set `activity_id` to this custom string to define which Live Activity you want to update. |

Reviewer: ```suggestion
| `activity_id` | Required | String | When you register your Live Activity using [`launchActivity`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class), you use the `pushTokenTag` parameter to name the Activity's push token to a custom string.<br><br>Set `activity_id` to this custom string to define which Live Activity you want to update. |
```

> | `content_state` | Required | Object | You define the `ContentState` parameters when you create your Live Activity. Pass the updated values for your `ContentState` using this object. The format of this request must match the shape you initially defined. |

Reviewer: ```suggestion
| `content_state` | Required | Object | You define the `ContentState` parameters when you create your Live Activity. Pass the updated values for your `ContentState` using this object.<br><br>The format of this request must match the shape you initially defined. |
```

| `end_activity` | Optional | Boolean | If `true`, this request ends the Live Activity. |
| `dismissal_date` | Optional | Datetime <br>([ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) string) | This parameter defines the time to remove the Live Activity from the user's UI. If this time is in the past, the Live Activity will be removed immediately. |
| `stale_date` | Optional | Datetime <br>([ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) string) | This parameter tells the system when the Live Activity content is marked as outdated in the user's UI. |
| `notification` | Optional | Object | Include an [`apple_push`]({{site.baseurl}}/api/objects_filters/messaging/apple_object/) object to define a push notification that will also be displayed with this update. |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3 .reset-td-br-4}

> ## Example Request

Reviewer: ```suggestion
## Example request
```

> <!--Is "apple_push" needed or helpful in the sample request below, or is it just clutter?-->

Reviewer 2: I think it's useful to show the example since it's documented just above.

Author: I *think* the code works without including the `apple_push`, and just going directly to `alert`, but would need someone to confirm.

Reviewer 3: Yeah, `apple_push` isn't needed here.

What I think should work is:
```
"notification": { "alert": {"body":"body","title":"title"}}
```

(but I'm not seeing it work in QA at the moment, @jerielng is looking into expected behavior)

```json
curl --location --request POST 'https://rest.iad-01.braze.com/messages/live_activity/update \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer {YOUR-REST-API-KEY}' \
--data-raw '{
    "app_id": "{YOUR-APP-API-IDENTIFIER}",
    "activity_id": "live-activity-1",
    "content_state": {
        "teamOneScore": 2,
        "teamTwoScore": 4
    },
    "end_activity": false,
    "dismissal_date": "2023-02-28T00:00:00+0000",
    "stale_date": "2023-02-27T16:55:49+0000",
    "notification": {
        "apple_push": {
            "alert": "Halftime!",
            "badge": 1
        }
    }
}
```

## Response

<!--What are the status codes that can be returned for this endpoint? What troubleshooting is available for customers? -->

> There are XX status code responses for this endpoint: `201`, XX, and XX.

Reviewer 2: Are 201 and 400 only status codes returned?

Reviewer 3: Yeah, those are the only ones that need documenting, I think.

It's possibly to get some 5xx responses as well if things are broken on our end :P

Author: I think 5XX errors are mentioned in other API docs. I'll go ahead and throw 'em in.

### Response parameters

<!--Are there any response params worth mentioning, or are they all messages from the server?-->

### Example success response

> The status code `201` could return the following response body.

Reviewer 3: It may be worth pointing out that this only means the request was well-formed and we received it. If, for example, they get the activity_id value wrong, it'll still respond with success even though nothing will be sent.

```json
{
  "message": "success"
}
```

### Example error response

The status code `400` could return the following response body. Refer to the [API errors and responses article]({{site.baseurl}}/api/errors/) for more information about errors you may encounter.

<!--Can you please include an example error response here? -->

```json

> 

Reviewer 3: Here's one example, if the content isn't valid JSON:
```
{
  "error": "\nProblem:\n  message body does not match declared format\nResolution:\n  when specifying application/json as content-type, you must pass valid application/json in the request's 'body' "
}
```

Author: Thanks so much!

---
nav_title: "POST: Update Live Activity"
article_title: "POST: Update Live Activity"
search_tag: Endpoint

> page_order: 1

Author: To do note to self: Add hidden tag
